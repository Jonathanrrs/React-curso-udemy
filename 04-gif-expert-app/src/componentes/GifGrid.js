import React, { useState, useEffect } from 'react'

export const GifGrid = ({category}) => {

    const [count, setCount] = useState(0);

    useEffect(() => {
        getGifs();
    }, [])

    const getGifs = async() => {
        const url = `https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=vD2873pCOTBaVMmRB83BYa1KkLi62PhF`
        const resp = await fetch(url);
        const {data} = await resp.json();
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(gifs);
    }

    return (
        <div>
            <h3>{category}</h3>
            <h1>{count}</h1>
            <button onClick={() => setCount(count+1)}></button>
        </div>
    )
}
