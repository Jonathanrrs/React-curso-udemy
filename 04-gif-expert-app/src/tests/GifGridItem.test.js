import React from 'react';
import {shallow} from 'enzyme';
import { GifGridItem } from '../componentes/GifGridItem';

describe('Pruebas en GifGridItem', () => {
    
    test('Mostrar GifGridItem correctamente ', () => {
        
        const wrapper = shallow(<GifGridItem />);
        expect(wrapper).toMatchSnapshot();
    })
    
})
