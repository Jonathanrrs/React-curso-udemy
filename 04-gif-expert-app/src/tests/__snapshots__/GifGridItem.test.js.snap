// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Mostrar GifGridItem correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <p />
</div>
`;
